### PR TITLE
bump to 0.4.4

### PR DESCRIPTION
**Pull request by mastoca**
attempt at fixing rocm. I'm still building cuda which I expect to work but will leave comments if things get bumpy.

### CODE REVIEW

**Comment by mastoca** [comment]
both platforms now build and cuda is tested and works.

rocm builds not sure on how well it runs.

File: package.nix [hunk 1]
```diff
@@ -180,7 +185,8 @@ goBuild {
 
   preBuild = ''
     # build llama.cpp libraries for ollama
-    make -j 5
+    threads="$(nproc)"
+    make -j $((threads / 2))
```
Comment (abysssol) [inline package.nix]: I'm now starting to wonder whether `threads / 2` is too low. Do you think it should be higher?

```suggestion
    make -j $((threads * 3 / 4))
```

File: package.nix [hunk 2]
```diff
@@ -166,6 +166,11 @@ goBuild {
     ++ lib.optionals enableCuda cudaLibs
     ++ lib.optionals stdenv.hostPlatform.isDarwin metalFrameworks;
 
+  patches = [
+    # sin-eating: ollama's build script is unable to find hipcc
```
Comment (abysssol) [inline package.nix:170]: I'm not familiar with the term "sin-eating". Does it have any technical meaning? I looked at the [sin-eater wikipedia article](https://en.wikipedia.org/wiki/Sin-eater), but I feel like I still don't get what you meant.

**Review by abysssol — APPROVED** [review]
Rocm works correctly! I measured GPU utilization with `radeontop`, and tested `tinydolphin` and `llama3.2` (which are both small).

Side note: I do (still) need to set `HSA_OVERRIDE_GFX_VERSION=10.3.0`.

File: package.nix (hunk 2 above)
Comment (mastoca) [reply package.nix:170]: it was a reference to nixcon 2024 where one of the presenters referred to nixpkgs as sin-eaters of linux app packaging.

**Comment by abysssol** [comment]
Thank you for figuring out how to fix rocm!

**Pull request merged**